src/neuroglancer/datasource/computed/frontend.ts

```diff
@@ -0,0 +1,252 @@
+  inputSourceIndex: number;
+  // Output voxel size.
+  outputVoxelSize: vec3;
+  // Output voxel offset.
+  outputVoxelOffset: vec3;
+  // Output volume size.
+  outputVolumeSize: vec3;
+  // Determines input buffer size as well as output data chunks size, etc.
+  computationParameters: ComputationParameters;
+
+  dataSourceProvider: DataSourceProvider;
+}
+
+interface ComputedVolumeSpecs {
+  sources: VolumeChunkSource[][];
+  volumeType: VolumeType;
+  dataType: DataType;
+  numChannels: number;
+}
+
+export class ComputedMultiscaleVolumeChunkSource implements GenericMultiscaleVolumeChunkSource {
+  numChannels: number;
+  dataType: DataType;
+  volumeType: VolumeType;
+
+  constructor(
+      public params: ComputedVolumeDataSourceParameters, public sources: VolumeChunkSource[][],
+      public computation: VolumeComputationFrontend, public chunkManager: ChunkManager) {
+    this.numChannels = params.computationParameters.outputSpec.numChannels;
+    this.dataType = params.computationParameters.outputSpec.dataType;
+    this.volumeType = params.computationParameters.outputSpec.volumeType;
+  }
+
+  getSources(volumeSourceOptions: VolumeSourceOptions) {
+    const spec = VolumeChunkSpecification.getDefaults({
+      voxelSize: this.params.outputVoxelSize,
+      dataType: this.dataType,
+      numChannels: this.numChannels,
+      chunkDataSizes: [this.params.computationParameters.outputSpec.size],
+      transform: mat4.fromTranslation(
+          mat4.create(),
+          vec3.multiply(vec3.create(), this.params.outputVoxelSize, this.params.outputVoxelOffset)),
+      upperVoxelBound: this.params.outputVolumeSize,
+      volumeType: this.volumeType,
+      baseVoxelOffset: this.params.outputVoxelOffset,
+      volumeSourceOptions,
+    })[0];
+
+    const originSource = this.sources[this.params.inputScaleIndex][this.params.inputSourceIndex];
+    const inputResolution = originSource.spec.voxelSize;
+
+    const parameters: ComputedVolumeChunkSourceParameters = {
+      computationRef: this.computation.addCounterpartRef(),
+      sourceRef: originSource.addCounterpartRef(),
+      inputSize: this.params.computationParameters.inputSpec.size,
+      scaleFactor: vec3.divide(vec3.create(), this.params.outputVoxelSize, inputResolution)
+    };
+
+    const computedSource =
+        this.chunkManager.getChunkSource(ComputedVolumeChunkSource, {spec, parameters});
+
+    return [[computedSource]];
+  }
+
+  getMeshSource() {
+    return null;
+  }
+}
+
+
+export class ComputedDataSource extends DataSource {
+  getOriginVolumes(
+      dataSourceProvider: DataSourceProvider, originUrl: string, chunkManager: ChunkManager,
+      cancellationToken: CancellationToken): Promise<ComputedVolumeSpecs> {
+    const slashPosition = originUrl.indexOf('://');
+    const config = originUrl.substring(slashPosition + 3);
+    return new Promise((resolve, reject) => {
+             const dataSource = dataSourceProvider.getDataSource(originUrl)[0];
```

Comment:
You can just call dataSourceProvider.getVolume

Reply:
Done.